toast-app/src/hooks/useToastContainer.js

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react'
+import { toastService } from '@/ToastService'
+
+export const useToastContainer = () => {
+  const [toastList, setToastList] = useState(
+    toastService.toastList,
+  )
+
+  const updateList = (list) => {
+    setToastList(list)
+  }
+
+  useEffect(() => {
+    toastService.subscribe('useToastContainer', updateList)
```

Comment:
Move 'useToastContainer' to the constant and rename to 'DISPATCH_UPDATE_LIST', for example.